File: app/src/main/java/nejati/me/service/part/HealthService.java
package nejati.me.service.part;

import org.jetbrains.annotations.NotNull;

import nejati.me.service.generator.ServiceGenerator;
import nejati.me.service.listener.OnServiceStatus;
import nejati.me.service.model.healthStatusModel.response.HealthStatusResponse;

/**
 * Authors:
 * Reza Nejati <reza.n.j.t.i@gmail.com>
 * Copyright © 2017
 */
public class HealthService  extends BasePart {
    public HealthService(ServiceGenerator serviceGenerator) {
        super(serviceGenerator);
    }

    @NotNull
    @Override
    protected BasePart getPart() {
        return this;
    }
    public void healthService(OnServiceStatus<HealthStatusResponse> listener) {
        start(getServiceGenerator().createService().health(), listener);
    }
}

File: app/src/main/java/nejati/me/service/part/HealthService.kt
package nejati.me.service.part

import java.util.ArrayList

import nejati.me.service.generator.ServiceGenerator
import nejati.me.service.listener.OnServiceStatus
import nejati.me.service.model.healthStatusModel.response.HealthStatusResponse
import nejati.me.service.model.questionsModel.response.QuestionsResponse

class HealthService(serviceGenerator: ServiceGenerator) : BasePart(serviceGenerator) {

    override val part: BasePart
        get() = this

    /**
     * @param listener
     */
    fun healthStatusService(listener: OnServiceStatus<HealthStatusResponse>) {
        start(serviceGenerator.createService().healthStatus(), listener)
    }

    /**
     * @param listener
     * @param questionFilter
     */
    fun questionsListService(
        listener: OnServiceStatus<ArrayList<QuestionsResponse>>,
        questionFilter: String) {
        start(serviceGenerator.createService().questions(questionFilter), listener)
    }


    /**
     * @param listener
     * @param questionId Id Of Question
     */
    fun questionsDetailService(
        listener: OnServiceStatus<ArrayList<QuestionsResponse>>,
        questionId: Int) {
        start(serviceGenerator.createService().questionsDetail(questionId), listener)
    }
}